timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockEventLogImpl.java

```diff
@@ -37,6 +41,14 @@
 import java.util.stream.Collectors;
 
 public class LockEventLogImpl implements LockEventLog {
+    private static final SafeLogger log = SafeLoggerFactory.get(LockEventLogImpl.class);
+
+    // Our Sliding window can hold a maximum of 1000 values. If we never store more than 100 metadata values at a time
+    // memory usage by metadata will be bounded by 100 * 1000 * MAX_CHANGE_METADATA bytes.
+    // Assuming we are using ChangeMetadata that holds ~2KB max each, metadata cannot take up more than 200MB of memory.
+    @VisibleForTesting
+    static final int MAX_METADATA_SIZE = 100;
```

Comment:
I wanted metadata to consume less than 1GB at worst, so 100 was the closest power of ten. Should we increase this?